File: javajava/Aleatorio.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#include <stdbool.h>

void processInput();
void Regerar(char *palavra, int i);
bool Comparar(const char *palavra);

int main() {
    processInput();
    printf("FIM\n");
    return 0;
}

void processInput() {
    char palavra[100];
    srand(4); 
    fgets(palavra, sizeof(palavra), stdin);
//Tudo o que estaria na main está aqui para facilitar a chamada do método nele mesmo.
    if (Comparar(palavra)) {
        return;
    } else {//Se a palavra for FIM, ele se encerra, se não, continua indo.
        Regerar(palavra, 0); 
        printf("%s", palavra); 
        processInput();
    }
}

void Regerar(char *palavra, int i) {
    int tamanho = strlen(palavra);
    if (i < tamanho - 1) { 
        char letra1 = 'a' + (rand() % 26);
        char letra2 = 'a' + (rand() % 26);
        if (palavra[i] == letra1) {
            palavra[i] = letra2;
        }//Duas letras são sorteadas, caso a primeira apareça, é substituída pela segunda.
        i++;
        Regerar(palavra, i);
    }
}

bool Comparar(const char *palavra) {//Compara se a palavra é igual a FIM.
    return strcmp(palavra, "FIM\n") == 0; 
}

File: javajava/Aleatorio.java
import java.util.Random;
public class Aleatorio{
    public static void main(String[] args){
        Random gerador= new Random();
        gerador.setSeed(4);
        String palavra=MyIO.readLine();
        while(Comparar(palavra)==false){
        Regerar(palavra, gerador);
        System.out.print("\n");
        palavra=MyIO.readLine();
        }
        MyIO.println("FIM");
    }

    public static boolean Comparar(String palavra){
        String palavra2="FIM";
        if(palavra.length()!=palavra2.length()){//Compara a palavra mandada letra por letra, e pelo tamanho para identificar se a mesma é FIM.
            return false;
        }
        else{
        for(int i=0; i<palavra.length(); i++){
            if(palavra.charAt(i)!=palavra2.charAt(i)){
            return false;
            }
        }
      }
      return true;
    }

    public static void Regerar(String palavra, Random gerador){
        char letra1=((char)('a' + (Math.abs(gerador.nextInt()) % 26)));
        char letra2=((char)('a' + (Math.abs(gerador.nextInt()) % 26)));
        char temp;
        //Gera duas letras aleatórias por meio do gerador, e troca todas as ocorrências da primeira letra pela segunda.
        for(int i=0; i<palavra.length(); i++){
            if(palavra.charAt(i)==letra1){
            temp=letra1;
            letra1=letra2;
            letra2=temp;
            MyIO.print(letra1);
            }
            else{
                MyIO.print(palavra.charAt(i));
            }
        }
    }
}